src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -1152,9 +1152,13 @@ public StringBuilder Remove(int startIndex, int length)
 
         public StringBuilder Append(char value)
         {
-            if (m_ChunkLength < m_ChunkChars.Length)
+            int nextCharIndex = m_ChunkLength;
+            char[] chars = m_ChunkChars;
+
+            if ((uint)chars.Length > (uint)nextCharIndex)
             {
-                m_ChunkChars[m_ChunkLength++] = value;
+                chars[nextCharIndex] = value;
+                m_ChunkLength++;
```

Comment:
```suggestion
                m_ChunkLength = nextCharIndex + 1;
```
`nextCharIndex` is already in a register.

Reply:
@adamsitnik isn't this change it worth or did you just oversee it?

```diff
- inc dword ptr [rsi + 0x18]
+ inc ecx
+ mov dword ptr [rsi + 0x18], ecx
```
Saves one memory load.

Reply:
Hmm, it may be worth taking a look why the JIT doesn't do it itself. Seems like it could/should.

Reply:
@gfoidl I had no idea that this change can affect the code gen (I would hope the JIT to do it for me). I currently have a long list of potentially bigger perf improvements for `StringBuilder` and I am going to focus on these. Please feel free to experiment and send a PR if you can prove that improves the perf.

Thanks!

Reply:
Turns out that the JIT does make the extra load a CSE candidate but it doesn't consider it worthwhile enough to actually perform the CSE. Oh well, the CSE heuristic may need tuning but that's not trivial.